conftest.py

```diff
@@ -1,5 +1,133 @@
+            self._tw.write(extra, **kwargs)
+            self.currentfspath = -2
+
+    def write_path_name(self, nodeid):
+        # called from `pytest_runtest_logstart`:
+        # responsible for writing bold path name above list of all tests for that file
+        pathname = nodeid.split("::")[0]
+        if pathname not in self.testfiles:
+            self.testclasses = []
+            if self.testfiles != []:
+                self._tw.line()
+            self.testfiles.append(pathname)
+            self.write_fspath_result(pathname, "", **({'bold': True}))
+
+    def write_class_name(self, nodeid):
+        # called from `pytest_runtest_logstart`:
+        # responsible for writing class name above list of all tests for that class
+        if len(nodeid.split("::")) >= 3:
+            classname = nodeid.split("::")[1]
+            if classname not in self.testclasses:
+                self.testclasses.append(classname)
+                self.write_fspath_result("  {}".format(classname), "")
+
+    def pytest_runtest_logstart(self, nodeid, location):
+        # inbuilt pytest reporter method; extended with `write_path_name` and `write_class_name`
+        if self.showlongtestinfo:
+            line = self._locationline(nodeid, *location)
+            self.write_ensure_prefix(line, "")
+        elif self.showfspath:
+            self.write_path_name(nodeid)
+            self.write_class_name(nodeid)
+
+    def pytest_runtest_logreport(self, report):
+        # inbuilt pytest reporter method; changes made:
+        # test_title introduced. It formats test title to make it more readable
+        # checking for verbosity is turned off (verbose by default) as this plugin works best for verbose mode.
+        # indentation and markup for test title introduced
+        rep = report
+        test_title = (rep.nodeid.split("::")[-1]).split("_")
+        if test_title[0].lower() == "test":
+            test_title.pop(0)
+        test_title = " ".join(test_title)
+        test_title += " "
+        res = self.config.hook.pytest_report_teststatus(report=report)
+        cat, letter, word = res
+        self.stats.setdefault(cat, []).append(rep)
+        self._tests_ran = True
+        if not letter and not word:
+            # probably passed setup/teardown
+            return
+        # if self.verbosity <= 0:
+        #     if not hasattr(rep, 'node') and self.showfspath:
+        #         self.write_fspath_result(rep.nodeid, letter)
+        #     else:
+        #         self._tw.write(letter)
+        # else:
+        if isinstance(word, tuple):
+            word, markup = word
+        else:
+            if rep.passed:
+                markup = {'green':True}
+            elif rep.failed:
+                markup = {'red':True}
+            elif rep.skipped:
+                markup = {'yellow':True}
+        line = self._locationline(rep.nodeid, *rep.location)
+        if not hasattr(rep, 'node'):
+            word = "({})".format(word)
```

Comment:
A tiny recommendation would be to avoid repeating the `self.write_ensure_prefix` calls by assigning the `"  {}".format(test_title)` to a variable in the `if` statement below and then calling the method afterwards. That way, if you ever have to change your method call, it only needs to be done in one place.